### DISCUSSION TITLE
Encrypting

### DISCUSSION THREAD

**Discussion by yannickgloster** [nextauthjs/next-auth#1831] [Help]
The docs say this:

> Auto-generates symmetric signing and encryption keys for developer convenience

But there doesn't seem to be any way to use the keys for encrypting other files.

**Comment by StefanSelfTaught** [comment]
I don't think there is a way to get the encryption key yet. You can generate yourself one pretty easily and use it.

**Reply by balazsorban44** [reply]
As the docs says, it is there
>  for developer convenience

and nothing more. In general, if you want to use the secret, you should really provide your own keys and generate them in a secure way.

You can try using `jose newkey -s 512 -t oct -a HS512` (https://www.npmjs.com/package/node-jose-tools)

https://next-auth.js.org/configuration/options#json-web-token-options

Also, what do you mean by "other files"?

**Reply by yannickgloster** [reply]
Ah so just for a bit of context, I'm working on a project where users need to have RSA key pairs to securely store files. I can generate them fine but I would like the private key to be stored securely behind the user's password. I am using OAuth too which probably doesn't make it any easier.

I am all new to this authentication stuff and it's just for a college project so it's not big deal.

**Reply by yannickgloster** [reply]
I will say, very cool library though

**Reply by yannickgloster** [reply]
@balazsorban44 would there be any way to have my generated private key encrypted behind the user's password?

**Reply by balazsorban44** [reply]
Honestly, I don't know. :see_no_evil:

cc: @iaincollins
